File: Jongryul/03/11060.js
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";
const input = fs.readFileSync(filePath).toString().trim().split("\n");
const n = Number(input[0]);
const arr = input[1].split(" ").map(Number);
const dp = Array(n).fill(1001);
dp[0] = 0;
for (let i = 1; i < arr[0] + 1; i++) {
  dp[i] = 1;
}

for (let i = 1; i < n; i++) {
  for (let j = 1; j < arr[i] + 1; j++) {
    const k = i + j;
    if (k < n) {
      dp[k] = Math.min(dp[k], dp[i] + 1);
    }
  }
}
if (dp[n - 1] != 1001) {
  console.log(dp[n - 1]);
} else {
  console.log(-1);
}

File: 11st_Competition/Jongryul/03/11060.py
import sys
input = sys.stdin.readline
n = int(input())
arr = list(map(int, input().split()))
dp = [1e9] * n
dp[0] = 0
for i in range(1, arr[0]+1):
    if i < n:
        dp[i] = 1

for i in range(1, n):
    for j in range(1, arr[i]+1):
        k = i + j
        if k < n:
            dp[k] = min(dp[k], dp[i] + 1)
if dp[n-1] != 1e9:
    print(dp[n-1])
else:
    print(-1)
